ArduCopter/radio.cpp

```diff
@@ -139,7 +139,7 @@ void Copter::set_throttle_and_failsafe(uint16_t throttle_pwm)
     if (throttle_pwm < (uint16_t)g.failsafe_throttle_value) {
 
         // if we are already in failsafe or motors not armed pass through throttle and exit
-        if (failsafe.radio || !(ap.rc_receiver_present || motors.armed())) {
+        if (failsafe.radio || !(ap.rc_receiver_present || hal.util->get_soft_arm_state() != AP_HAL::Util::SOFT_ARM_STATE_DISARMED)) {
```

Comment:
You are absolutely right, I forgot to invert the OR, sorry about that. The parenthesis should be kept for better readability, but I don't think they are actually needed, because AND has precedence over OR in C++.
I think the version you wrote in this comment to be the more readable.
And yes, I also think that an updated comment would help.